plugins/configs/api/plugin.go

```diff
@@ -0,0 +1,76 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/configs"
+)
+
+type ConfigAPIPlugin struct {
+	ServiceAddress string               `toml:"service_address"`
+	Storage        config.StoragePlugin `toml:"storage"`
```

Comment:
Can you provide an example how to configure this? Looking into running Telegraf to see how the config api should run. So far this is what I could figure out for a config:
```
[[config.api]]
    storage = "internal"
```
Is there any documentation on how to configure Telegraf to use config-api?

Reply:
@helenosheaa helped shown an example for this:
```
[[config.api]]
  service_address = ":7551"
  [config.api.storage.internal]
    file = "config_state.db"
```

Probably good idea to add this into some documentation

Reply:
resolved in 06d4d3fb